Analytical_Engine/Query/Geometry.cs

```diff
@@ -155,7 +155,7 @@ public static IEnumerable<Face> Geometry<TFace>(this IEnumerable<TFace> faces)
         [Output("face", "The geometry of the IFace as geometrical Mesh Face.")]
         public static Face Geometry(this IFace face)
         {
-            if (face == null)
+            if (face == null || face.NodeListIndices == null)
```

Comment:
This would possibly be more consistent with other changes
```suggestion
            if (face?.NodeListIndices == null)
```